File: tests/benchmark/vulnerable/java/sqli_001.java
// BENCHMARK: vulnerable - sqli
import javax.servlet.http.*;
import java.sql.*;

public class UserServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws Exception {
        String userId = request.getParameter("id");
        Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/db", "user", "pass");
        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery("SELECT * FROM users WHERE id=" + userId);
        while (rs.next()) {
            response.getWriter().println(rs.getString("username"));
        }
        conn.close();
    }
}

File: tests/benchmark/vulnerable/python/sqli_001.py
# BENCHMARK: vulnerable - sqli
from flask import Flask, request
import sqlite3

app = Flask(__name__)


@app.route("/user")
def get_user():
    uid = request.args.get("id")
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    cursor.execute(f"SELECT * FROM users WHERE id={uid}")
    row = cursor.fetchone()
    conn.close()
    return str(row)
